lib/less/tree/call.js

```diff
@@ -29,8 +29,15 @@ Call.prototype.accept = function (visitor) {
 // The function should receive the value, not the variable.
 //
 Call.prototype.eval = function (context) {
-    var args = this.args.map(function (a) { return a.eval(context); }),
-        result, funcCaller = new FunctionCaller(this.name, context, this.getIndex(), this.fileInfo());
+    /* Used for calc() */
+    if (this.mathOff) {
+        context.mathOn = false;
+    }
+    var args = this.args.map(function (a) { return a.eval(context); });
+    if (this.mathOff) {
+        context.mathOn = true;
+    }
+    var result, funcCaller = new FunctionCaller(this.name, context, this.getIndex(), this.fileInfo());
     
```

Comment:
Hmm, (sorry I can't really test it - so this is abstract code reading from me), the call should set `mathOn=true`, evaluate args and then *restore* prev. mathOn value. I.e. in summary (counting my other coment above), pseudocode:
```js
Call.prototype.eval = function (context) {
    ...
    var prevMath = context.mathOn;
    if (funcName === "calc") 
        context.mathOn = false;

    var args = this.args.map(function (a) { return a.eval(context); });

    context.mathOn = prevMath;

    ...
}
```

And the test to ensure:
```less
foo: 1 + 2 calc(3 + 4) 5 + 6; // -> 3 calc(3 + 4) 11;
```

Reply:
You're absolutely right.

Reply:
Fixed and test added!